### PR TITLE
Filesystem minor consistency

### PR DESCRIPTION
**Pull request by mppf**
Change the FileSystem module to:
   1) Use always error= as the error argument
   2) Rename viewMode to getMode (like getUID or getGID)

These changes are for consistency.

### CODE REVIEW

**Comment by mppf** [comment]
@lydia-duncan can you review? I'm testing now.

File: modules/standard/FileSystem.chpl [hunk 1]
```diff
@@ -729,7 +729,7 @@ proc umask(mask: int): int {
    err: a syserr used to indicate if an error occurred during this function
    name: the name of the file that you want to know the permissions of.
 */
-proc viewMode(out err: syserr, name: string): int {
+proc getMode(out error: syserr, name: string): int {
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: Would you mind re-alphabetizing this?

File: modules/standard/FileSystem.chpl [hunk 2]
```diff
@@ -27,7 +27,7 @@ use Error;
    Note: this is not safe within a parallel context.  A chdir call in one task
    will affect the current working directory of all tasks for that locale.
 */
-proc locale.chdir(out err: syserr, name: string) {
+proc locale.chdir(out error: syserr, name: string) {
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: The use of the argument will probably need to be updated (though I'm sure testing will tell you that shortly)

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: This goes for all of them, it looks like.

Comment (mppf) [inline modules/standard/FileSystem.chpl]: The uses of the argument within this file are OK since they go by order.

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: So err = blah, etc. is still fine in those functions?  That seems confusing.

Comment (mppf) [inline modules/standard/FileSystem.chpl]: The variable passed doesn't have to have the same name as the argument. We can fix that later if you want - I just wanted the public interface to be set for code freeze. Still waiting on tests...

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: Sorry, I'm not being clear.  The uses of the argument within the function itself have not been updated.

File: modules/standard/FileSystem.chpl [hunk 3]
```diff
@@ -59,7 +59,7 @@ proc locale.chdir(name: string) {
               question.  See description of the provided constants for
               potential values.
 */
-proc chmod(out err: syserr, name: string, mode: int) {
+proc chmod(out error: syserr, name: string, mode: int) {
   extern proc chpl_fs_chmod(name: c_string, mode: int): syserr;
 
   err = chpl_fs_chmod(name.c_str(), mode);
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: For instance, here.

Comment (mppf) [inline modules/standard/FileSystem.chpl]: Right! I'll fix that

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: Let me know if testing didn't detect that (for file functions added before a couple of weeks ago).  I'm going to add more tests of all the new functions, and it'd be helpful for me to know if I need to add tests for one of the older ones.

**Comment by lydia-duncan** [comment]
Looks great!  Feel free to merge when testing completes

**Comment by mppf** [comment]
@lydia-duncan I made the corresponding change in Path.chpl too, and now it passes all the tests that failed (test/modules/standard/FileSystem and test/modules/standard/Path and test/studies/filerator).

**Pull request merged**